mlflow/sagemaker/__init__.py

```diff
@@ -1336,6 +1336,22 @@ def _get_sagemaker_config_name(endpoint_name):
     return f"{endpoint_name}-config-{get_unique_resource_id()}"
 
 
+def _get_sagemaker_config_tags(endpoint_name):
+    return [{"Key": "app_name", "Value": endpoint_name}]
+
+def _prepare_sagemaker_tags(
+  sagemaker_tags: Dict[str, str],
+  config_tags: List[Dict[str, str]]  
+):
+    """
+    Convert dict of tags to list for SageMaker resources and adds to config tags list.
+    """
+    if sagemaker_tags:
+        sagemaker_tags = [{"Key": key, "Value": str(value)} for key, value in sagemaker_tags.items()]
+        config_tags.extend(sagemaker_tags)
+
+    return sagemaker_tags
```

Comment:
Would something like this perhaps be a little bit more robust?

```python
from mlflow import MlflowException
from typing import List, Dict, Optional

SAGEMAKER_APP_NAME_TAG_KEY = "app_name"

def _get_sagemaker_config_tags(endpoint_name):
    return [{"Key": SAGEMAKER_APP_NAME_TAG_KEY, "Value": endpoint_name}]

def _prepare_sagemaker_tags(
    config_tags: List[Dict[str, str]],
    sagemaker_tags: Optional[Dict[str, str]]=None,
):
    if not sagemaker_tags:
        return config_tags
    
    if SAGEMAKER_APP_NAME_TAG_KEY in sagemaker_tags:
        raise MlflowException.invalid_parameter_value(f"Duplicate tag provided for '{SAGEMAKER_APP_NAME_TAG_KEY}'")
    parsed = [{"Key": key, "Value": str(value)} for key, value in sagemaker_tags.items()]

    return config_tags + parsed
```

Validating the behavior: 
```python
_prepare_sagemaker_tags([{"Key": "app_name", "Value": "my_app"}], {"a": "1", "b": "2", "c": "3"})
```
```shell
[{'Key': 'app_name', 'Value': 'my_app'},
 {'Key': 'a', 'Value': '1'},
 {'Key': 'b', 'Value': '2'},
 {'Key': 'c', 'Value': '3'}]
```

```python
_prepare_sagemaker_tags([{"Key": "app_name", "Value": "my_app"}], {"app_name": "my_better_app", "b": "2", "c": "3"})
```
```shell
---------------------------------------------------------------------------
MlflowException                           Traceback (most recent call last)
/var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/ipykernel_12446/3588563336.py in <cell line: 1>()
----> 1 _prepare_sagemaker_tags([{"Key": "app_name", "Value": "my_app"}], {"app_name": "my_better_app", "b": "2", "c": "3"})

/var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/ipykernel_12446/3590447905.py in _prepare_sagemaker_tags(config_tags, sagemaker_tags)
     14 
     15     if SAGEMAKER_APP_NAME_TAG_KEY in sagemaker_tags:
---> 16         raise MlflowException.invalid_parameter_value(f"Duplicate tag provided for '{SAGEMAKER_APP_NAME_TAG_KEY}'")
     17     parsed = [{"Key": key, "Value": str(value)} for key, value in sagemaker_tags.items()]
     18 

MlflowException: Duplicate tag provided for 'app_name'
```
```python
_prepare_sagemaker_tags([{"Key": "app_name", "Value": "my_app"}], None)
_prepare_sagemaker_tags([{"Key": "app_name", "Value": "my_app"}], {})
```
```shell
[{'Key': 'app_name', 'Value': 'my_app'}]
```

Does this reflect the intended functionality for the ``_prepare_sagemaker_tags()`` function?

Reply:
Yep, this looks great.